src/cwe_checker_lib/src/analysis/function_signature/mod.rs

```diff
@@ -305,4 +422,40 @@ pub mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_parameter_merging() {
+        let mut func_sig = FunctionSignature::mock_x64();
+        let stack_parm_1 = Arg::Stack {
+            address: expr!("RSP:8 + 0x1000:8"),
+            size: 8.into(),
+            data_type: Some(Datatype::Integer),
```

Comment:
This is just nitpicking, but I think we should leave the data types as `None` even in the unit tests. It just should never occur that we already have some data type information in the function signature analysis.